akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -233,6 +233,24 @@ private[http] object StreamUtils {
       def open(): Unit = promise.success(())
     }
   }
+
+  /**
+   * Returns a flow that is almost identity but doesn't pass cancellation from downstream to upstream.
+   *
+   * Note: This might create a resource leak if the upstream never completes. External measures
+   * need to be taken to ensure that the upstream will always complete.
+   */
+  def ignoreCancellation[T]: Flow[T, T, NotUsed] = Flow.fromGraph(new IgnoreCancellation)
+  class IgnoreCancellation[T] extends SimpleLinearGraphStage[T] {
```

Comment:
@jrudolph  Apologies for the seemingly dumb question, but doesn't *ignoring cancellation* violate Reactive Streams?
(I'm more than likely missing something here?) Also public class and not final?